File: Contests/16/D.py
t = int(input())
for _ in range(t):
    n = int(input())
    s = input()
    
    if n % 2 == 1:
        print("-1")

    else:
        res = []
        left_cnt = 0
        if s[0] == '(':
            current = 1 
        else:
            current = 2
        for c in s:
            if c == '(':
                left_cnt += 1
                res.append(str(current))
                current = 1 if current != 1 else 2
            else:
                res.append(str(current))
                current = 1 if current != 2 else 2
        
        if left_cnt != n // 2:
            print(-1)
        else:
            print('2' if current == 2 else '1')
            print(' '.join(res))

File: Contests/16/D.cpp
#include <iostream>
#include <string>
#include <vector>

int main() {
    int t;
    std::cin >> t;

    while (t--) {
        int n;
        std::cin >> n;
        
        std::string s;
        std::cin >> s;

        if (n % 2 == 1) {
            std::cout << "-1" << std::endl;
        } else {
            std::vector<std::string> res;
            int left_cnt = 0;
            int current = (s[0] == '(') ? 1 : 2;

            for (char c : s) {
                if (c == '(') {
                    left_cnt++;
                    res.push_back(std::to_string(current));
                    current = (current != 1) ? 1 : 2;
                } else {
                    res.push_back(std::to_string(current));
                    current = (current != 2) ? 2 : 1;
                }
            }

            if (left_cnt != n / 2) {
                std::cout << "-1" << std::endl;
            } else {
                std::cout << ((current == 2) ? "2" : "1") << std::endl;
                for (const std::string& num : res) {
                    std::cout << num << " ";
                }
                std::cout << std::endl;
            }
        }
    }

    return 0;
}
